File: src/main/scala/com/github/emildafinov/ad/sdk/event/resolvers/EventResolverFactory.scala
package com.github.emildafinov.ad.sdk.event.resolvers

import com.github.emildafinov.ad.sdk.authentication.AppMarketCredentialsSupplier
import com.github.emildafinov.ad.sdk.event.EventResolverFactory
import com.github.emildafinov.ad.sdk.event.marshallers.EventResultMarshallersModule
import com.github.emildafinov.ad.sdk.{ClientDefinedCredentialsModule, ConnectorResolversAkkaDependenciesModule}

object EventResolverFactory {
  def apply(credSupplier: AppMarketCredentialsSupplier): EventResolverFactory = {
    val factoryApplicationContext = new ClientDefinedCredentialsModule
      with ConnectorResolversAkkaDependenciesModule
      with EventResultMarshallersModule
      with EventResolversModule {

      override protected val credentialsSupplier: AppMarketCredentialsSupplier = credSupplier
    }

    () => factoryApplicationContext.subscriptionOrderEventResolver
  }
}

File: src/main/java/com/github/emildafinov/ad/sdk/event/EventResolverFactory.java
package com.github.emildafinov.ad.sdk.event;

import com.github.emildafinov.ad.sdk.authentication.AppMarketCredentialsSupplier;
import com.github.emildafinov.ad.sdk.event.resolvers.EventResolverFactory$;
import com.github.emildafinov.ad.sdk.event.responses.SubscriptionOrderResponse;

/**
 * Used by SDK clients for accessing classes allowing to send back notifications about completed
 * events back to the monolith
 */
public interface EventResolverFactory {
	default EventResolverFactory get(AppMarketCredentialsSupplier credentialsSupplier) {
		return EventResolverFactory$.MODULE$.apply(credentialsSupplier);
	}
	EventResolver<SubscriptionOrderResponse> subscriptionOrderResolver(); 
}
